src/main/java/org/mule/service/http/impl/service/client/HttpMessageLogger.java

```diff
@@ -31,17 +34,24 @@ public HttpMessageLogger(final LoggerType loggerType) {
 
   @Override
   public void onDataReceivedEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
   @Override
   public void onDataSentEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
-  private void logBuffer(Buffer buffer) {
+  private static Map<String, String> getLogContext(Connection connection) {
+    Map<String, String> logContext = (Map<String, String>) connection.getAttributes().getAttribute("logContext");
```

Comment:
isn't the connection shared among different requests? (do you know this, @afelisatti ?) If that's the case, this will cause logging contexts to be mixed when concurrent requests occur